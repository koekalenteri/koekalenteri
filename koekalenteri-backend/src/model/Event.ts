export type Event = {
    id: String
    "event-type": String
    "classes": [String]
    "start-date": Date
    "end-date": Date
    "location": String
    "name": String
    "description": String
    "allow-online-entry": Boolean
    "allow-online-payment": Boolean
    "unofficial": Boolean
    "allow-owner-membership-priority": Boolean
    "allow-handler-membership-priority": Boolean
    "cost": Number
    "cost-member": Number
    "payment-details": String
    "account-number": String
    "reference-number": String
    "require-payment-before-entry": Boolean
    "judges": [Number]
    "official": Number   
    "created-at": String
    "created-by": String
    "modified-at": String
    "modified-by": String
}