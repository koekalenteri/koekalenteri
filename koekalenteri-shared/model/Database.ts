export interface JsonDbRecord {
  id: string
  createdAt: string
  createdBy: string
  deletedAt?: string
  deletedBy?: string
  modifiedAt: string
  modifiedBy: string
}

export interface DbRecord extends Omit<JsonDbRecord, 'createdAt' | 'modifiedAt' | 'deletedAt'> {
  createdAt: Date,
  modifiedAt: Date,
  deletedAt?: Date,
}

export interface JsonAuditRecord {
  auditKey: string
  timestamp: string
  user: string
  message: string
}

export interface AuditRecord extends Omit<JsonAuditRecord, 'timestamp'> {
  timestamp: Date
}
