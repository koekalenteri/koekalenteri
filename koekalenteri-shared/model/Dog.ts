import { Replace } from "./JSON";

export type BreedCode = '110' | '111' | '121' | '122' | '263' | '312';

export type JsonDog = {
  regNo: string
  name?: string
  rfid?: string
  breedCode?: BreedCode
  dob?: string
  gender?: DogGender
  refreshDate?: string
  results?: JsonTestResult[]
  titles?: string
  sire?: DogName,
  dam?: DogName
}

export type Dog = Replace<Replace<JsonDog, 'dob' | 'refreshDate', Date | undefined>, 'results', TestResult[] | undefined>;

export type DogName = {
  name?: string,
  titles?: string
}

export type DogGender = 'F' | 'M'

export type JsonTestResult = {
  type: string
  class: string
  date: string
  location: string
  result: string
  judge: string
  points?: number
  rank?: number
  ext?: string
  notes?: string
  cert?: boolean
  resCert?: boolean
}

export type TestResult = Replace<JsonTestResult, 'date', Date>
