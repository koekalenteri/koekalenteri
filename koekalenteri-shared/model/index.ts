export * from './utility'
export * from './JSON'
export * from './Database'

export * from './Dog'
export * from './Email'
export * from './Event'
export * from './EventType'
export * from './Person'
export * from './Organizer'
export * from './Registration'

export * from './paytrail'

export type Language = 'fi' | 'en'
