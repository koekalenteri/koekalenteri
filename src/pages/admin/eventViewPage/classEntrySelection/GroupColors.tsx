import type { RegistrationDate } from '../../../../types'

import Box from '@mui/material/Box'
import Stack from '@mui/material/Stack'

import GroupColorTooltip from './groupColors/GroupColorTooltip'

export const GROUP_COLORS = ['#2D9CDB', '#BB6BD9', '#F2994A', '#27AE60', '#828282', '#56CCF2']

export const availableGroups = (dates: Date[]) =>
  dates.reduce<RegistrationDate[]>((acc, date) => [...acc, { date, time: 'ap' }, { date, time: 'ip' }], [])

interface Props {
  readonly available: RegistrationDate[]
  readonly disableTooltip?: boolean
  readonly selected: RegistrationDate[]
}

const GroupColors = ({ available, selected, disableTooltip = false }: Props) => {
  if (disableTooltip) {
    const selIdx = available.findIndex(
      (dt) => selected[0].date.getTime() === dt.date.getTime() && selected[0].time === dt.time
    )
    return (
      <Box
        sx={{
          bgcolor: GROUP_COLORS[selIdx % GROUP_COLORS.length],
          width: 24,
          mr: '36px',
          height: '100%',
          borderTopLeftRadius: '4px',
        }}
      />
    )
  }
  return (
    <GroupColorTooltip selected={selected}>
      <Stack direction="row" spacing={0} sx={{ width: 36, height: '100%' }}>
        {available.map((dt, index) => {
          const color = GROUP_COLORS[index % GROUP_COLORS.length]
          const isSelected = !!selected.find((s) => s.date.getTime() === dt.date.getTime() && s.time === dt.time)
          return <Box key={color} sx={{ bgcolor: isSelected ? color : 'transparent', width: 6, height: '100%' }} />
        })}
      </Stack>
    </GroupColorTooltip>
  )
}

export default GroupColors
