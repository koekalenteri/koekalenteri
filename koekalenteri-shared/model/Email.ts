import { DbRecord, JsonDbRecord } from "./Database"

export type EmailTemplateId = 'registration'

export interface SESTemplate {
  TemplateName: string
  SubjectPart?: string
  TextPart?: string
  HtmlPart?: string
}

export interface EmailTemplate extends DbRecord {
  id: EmailTemplateId
  fi: string
  en: string
  ses?: {
    fi: SESTemplate
    en: SESTemplate
  }
}

export type JsonEmailTemplate = Omit<EmailTemplate, keyof DbRecord> & JsonDbRecord
