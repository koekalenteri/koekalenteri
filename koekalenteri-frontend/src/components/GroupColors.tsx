import { Box } from '@mui/material';
import { RegistrationDate } from 'koekalenteri-shared/model';

export const GROUP_COLORS = ['#2D9CDB', '#BB6BD9', '#F2994A', '#27AE60', '#828282', '#56CCF2'];

export const GroupColors = ({ dates, selected }: { dates: Date[]; selected: RegistrationDate[]; }) => {
  const available = dates.reduce<RegistrationDate[]>((acc, date) => [...acc, { date, time: 'ap' }, { date, time: 'ip' }], []);
  return <>{available.map((dt, index) => {
    const color = GROUP_COLORS[index % GROUP_COLORS.length];
    const isSelected = !!selected.find(s => s.date.getTime() === dt.date.getTime() && s.time === dt.time);
    return <Box key={color} sx={{ bgcolor: isSelected ? color : 'transparent', width: '6px', height: '100%' }} />;
  })}</>;
};
