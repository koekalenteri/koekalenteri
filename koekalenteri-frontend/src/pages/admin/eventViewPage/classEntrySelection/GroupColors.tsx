import React, { useMemo } from 'react'
import Box from '@mui/material/Box'
import Stack from '@mui/material/Stack'
import { RegistrationDate } from 'koekalenteri-shared/model'

import GroupColorTooltip from './groupColors/GroupColorTooltip'

export const GROUP_COLORS = ['#2D9CDB', '#BB6BD9', '#F2994A', '#27AE60', '#828282', '#56CCF2']

export const availableGroups = (dates: Date[]) =>
  dates.reduce<RegistrationDate[]>((acc, date) => [...acc, { date, time: 'ap' }, { date, time: 'ip' }], [])

interface Props {
  dates: Date[]
  disableTooltip?: boolean
  selected: RegistrationDate[]
}

const GroupColors = ({ dates, selected, disableTooltip = false }: Props) => {
  const available = useMemo(() => availableGroups(dates), [dates])
  if (disableTooltip) {
    const selIdx = available.findIndex(
      (dt) => selected[0].date.getTime() === dt.date.getTime() && selected[0].time === dt.time
    )
    return (
      <Box
        sx={{
          bgcolor: GROUP_COLORS[selIdx % GROUP_COLORS.length],
          width: 24,
          mr: '36px',
          height: '100%',
          borderTopLeftRadius: '4px',
        }}
      />
    )
  }
  return (
    <GroupColorTooltip selected={selected}>
      <Stack direction="row" spacing={0} sx={{ width: 36, height: '100%' }}>
        {available.map((dt, index) => {
          const color = GROUP_COLORS[index % GROUP_COLORS.length]
          const isSelected = !!selected.find((s) => s.date.getTime() === dt.date.getTime() && s.time === dt.time)
          return <Box key={color} sx={{ bgcolor: isSelected ? color : 'transparent', width: 6, height: '100%' }} />
        })}
      </Stack>
    </GroupColorTooltip>
  )
}

export default GroupColors
