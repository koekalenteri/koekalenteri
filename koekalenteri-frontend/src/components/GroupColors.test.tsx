import { render } from '@testing-library/react'

import { RegistrationDate } from '../../../koekalenteri-shared/model'

import { GroupColors } from './GroupColors'

describe('GroupColors', () => {

  const dates = [new Date(2020, 1, 1, 12), new Date(2020, 1, 2, 12), new Date(2020, 1, 3, 12)]
  const regDates = dates.reduce<RegistrationDate[]>((acc, date) => [...acc, {date, time: 'ap'}, {date, time: 'ip'}], [])
  const cases: [number, number, Date[], RegistrationDate[]][] = []

  for (let i = 0; i <= dates.length; i++) {
    const d = dates.slice(0, i)
    for (let j = 0; j <= regDates.length && j <= i * 2; j++) {
      cases.push([i, j, d, regDates.slice(0, j)])
    }
  }

  test.each(cases)('given %p dates and %p selected', (_i, _j, d, s) => {
    const { container } = render(<GroupColors dates={d} selected={s} />)
    expect(container).toMatchSnapshot()
  })
})
