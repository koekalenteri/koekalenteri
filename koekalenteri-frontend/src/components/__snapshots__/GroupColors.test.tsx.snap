// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupColors given 0 dates and 0 selected 1`] = `<div />`;

exports[`GroupColors given 1 dates and 0 selected 1`] = `
<div>
  <div
    class="MuiBox-root css-1urbi4i"
  />
  <div
    class="MuiBox-root css-1urbi4i"
  />
</div>
`;

exports[`GroupColors given 1 dates and 1 selected 1`] = `
<div>
  <div
    class="MuiBox-root css-14hq1a2"
  />
  <div
    class="MuiBox-root css-1urbi4i"
  />
</div>
`;

exports[`GroupColors given 1 dates and 2 selected 1`] = `
<div>
  <div
    class="MuiBox-root css-14hq1a2"
  />
  <div
    class="MuiBox-root css-1gwv2qh"
  />
</div>
`;

exports[`GroupColors given 2 dates and 0 selected 1`] = `
<div>
  <div
    class="MuiBox-root css-1urbi4i"
  />
  <div
    class="MuiBox-root css-1urbi4i"
  />
  <div
    class="MuiBox-root css-1urbi4i"
  />
  <div
    class="MuiBox-root css-1urbi4i"
  />
</div>
`;

exports[`GroupColors given 2 dates and 1 selected 1`] = `
<div>
  <div
    class="MuiBox-root css-14hq1a2"
  />
  <div
    class="MuiBox-root css-1urbi4i"
  />
  <div
    class="MuiBox-root css-1urbi4i"
  />
  <div
    class="MuiBox-root css-1urbi4i"
  />
</div>
`;

exports[`GroupColors given 2 dates and 2 selected 1`] = `
<div>
  <div
    class="MuiBox-root css-14hq1a2"
  />
  <div
    class="MuiBox-root css-1gwv2qh"
  />
  <div
    class="MuiBox-root css-1urbi4i"
  />
  <div
    class="MuiBox-root css-1urbi4i"
  />
</div>
`;

exports[`GroupColors given 2 dates and 3 selected 1`] = `
<div>
  <div
    class="MuiBox-root css-14hq1a2"
  />
  <div
    class="MuiBox-root css-1gwv2qh"
  />
  <div
    class="MuiBox-root css-wd5nlm"
  />
  <div
    class="MuiBox-root css-1urbi4i"
  />
</div>
`;

exports[`GroupColors given 2 dates and 4 selected 1`] = `
<div>
  <div
    class="MuiBox-root css-14hq1a2"
  />
  <div
    class="MuiBox-root css-1gwv2qh"
  />
  <div
    class="MuiBox-root css-wd5nlm"
  />
  <div
    class="MuiBox-root css-6m2gat"
  />
</div>
`;

exports[`GroupColors given 3 dates and 0 selected 1`] = `
<div>
  <div
    class="MuiBox-root css-1urbi4i"
  />
  <div
    class="MuiBox-root css-1urbi4i"
  />
  <div
    class="MuiBox-root css-1urbi4i"
  />
  <div
    class="MuiBox-root css-1urbi4i"
  />
  <div
    class="MuiBox-root css-1urbi4i"
  />
  <div
    class="MuiBox-root css-1urbi4i"
  />
</div>
`;

exports[`GroupColors given 3 dates and 1 selected 1`] = `
<div>
  <div
    class="MuiBox-root css-14hq1a2"
  />
  <div
    class="MuiBox-root css-1urbi4i"
  />
  <div
    class="MuiBox-root css-1urbi4i"
  />
  <div
    class="MuiBox-root css-1urbi4i"
  />
  <div
    class="MuiBox-root css-1urbi4i"
  />
  <div
    class="MuiBox-root css-1urbi4i"
  />
</div>
`;

exports[`GroupColors given 3 dates and 2 selected 1`] = `
<div>
  <div
    class="MuiBox-root css-14hq1a2"
  />
  <div
    class="MuiBox-root css-1gwv2qh"
  />
  <div
    class="MuiBox-root css-1urbi4i"
  />
  <div
    class="MuiBox-root css-1urbi4i"
  />
  <div
    class="MuiBox-root css-1urbi4i"
  />
  <div
    class="MuiBox-root css-1urbi4i"
  />
</div>
`;

exports[`GroupColors given 3 dates and 3 selected 1`] = `
<div>
  <div
    class="MuiBox-root css-14hq1a2"
  />
  <div
    class="MuiBox-root css-1gwv2qh"
  />
  <div
    class="MuiBox-root css-wd5nlm"
  />
  <div
    class="MuiBox-root css-1urbi4i"
  />
  <div
    class="MuiBox-root css-1urbi4i"
  />
  <div
    class="MuiBox-root css-1urbi4i"
  />
</div>
`;

exports[`GroupColors given 3 dates and 4 selected 1`] = `
<div>
  <div
    class="MuiBox-root css-14hq1a2"
  />
  <div
    class="MuiBox-root css-1gwv2qh"
  />
  <div
    class="MuiBox-root css-wd5nlm"
  />
  <div
    class="MuiBox-root css-6m2gat"
  />
  <div
    class="MuiBox-root css-1urbi4i"
  />
  <div
    class="MuiBox-root css-1urbi4i"
  />
</div>
`;

exports[`GroupColors given 3 dates and 5 selected 1`] = `
<div>
  <div
    class="MuiBox-root css-14hq1a2"
  />
  <div
    class="MuiBox-root css-1gwv2qh"
  />
  <div
    class="MuiBox-root css-wd5nlm"
  />
  <div
    class="MuiBox-root css-6m2gat"
  />
  <div
    class="MuiBox-root css-zea4zm"
  />
  <div
    class="MuiBox-root css-1urbi4i"
  />
</div>
`;

exports[`GroupColors given 3 dates and 6 selected 1`] = `
<div>
  <div
    class="MuiBox-root css-14hq1a2"
  />
  <div
    class="MuiBox-root css-1gwv2qh"
  />
  <div
    class="MuiBox-root css-wd5nlm"
  />
  <div
    class="MuiBox-root css-6m2gat"
  />
  <div
    class="MuiBox-root css-zea4zm"
  />
  <div
    class="MuiBox-root css-v7rqxr"
  />
</div>
`;
